python-connectors/google-calendar_events/connector.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from dataiku.connector import Connector
+from google_calendar_client import GoogleCalendarClient
+from dku_common import get_token_from_config
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO,
+                    format='google-calendar plugin %(levelname)s - %(message)s')
+
+
+class GoogleCalendarEventConnector(Connector):
+
+    def __init__(self, config, plugin_config):
+        Connector.__init__(self, config, plugin_config)
+        access_token = get_token_from_config(config)
+        self.client = GoogleCalendarClient(access_token)
+        self.from_date = self.config.get("from_date")
+        self.to_date = self.config.get("to_date", None)
+        self.calendar_id = self.config.get("calendar_id", "primary")
+        self.raw_results = self.config.get("raw_results", False)
+
+    def get_read_schema(self):
+        # In this example, we don't specify a schema here, so DSS will infer the schema
+        # from the columns actually returned by the generate_rows method
+        return None
+
+    def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
+                      partition_id=None, records_limit=-1):
```

Comment:
nitpick: If creating a `dku_constants.py` file, you can set `RECORDS_NO_LIMIT = -1` to make it cleaner, and then in the function definition
```python
generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                      partition_id=None, records_limit=constants.RECORDS_NO_LIMIT)
```